### PR TITLE
Sync refactor

### PR DESCRIPTION
**Pull request by almogdepaz**
## Motivation
readability and maintainability

## Changes
added a tryLock to simplify the code in syncer.go

## Test Plan
same test plan holds, no functionality change

### CODE REVIEW

**Comment by almogdepaz** [comment]
bors try

File: common/types/try_mutex.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import (
+	"sync"
+	"sync/atomic"
+	"unsafe"
+)
+
+const mutexLocked = 1 << iota
+
+// Mutex is simple sync.Mutex + ability to try to Lock.
+type TryMutex struct {
```
Comment (gavraz) [inline common/types/try_mutex.go:13]: Why not just use the package https://godoc.org/github.com/LK4D4/trylock
same implementation

Comment (almogdepaz) [reply common/types/try_mutex.go:13]: i did something a little more minimal then them also i don't want to be dependant on some small unknown repo

File: common/types/try_mutex.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package types
+
+import (
+	"sync"
+	"sync/atomic"
+	"unsafe"
+)
+
+const mutexLocked = 1 << iota
```
Comment (gavraz) [inline common/types/try_mutex.go]: remove iota

**Review by gavraz — APPROVED** [review]

**Comment by almogdepaz** [comment]
bors merge

**Pull request merged**